WeightedEuclidean.lua

```diff
@@ -25,55 +17,71 @@ function WeightedEuclidean:reset(stdv)
    if stdv then
       stdv = stdv * math.sqrt(3)
    else
-      stdv = 1./math.sqrt(self.templates:size(1))
+      stdv = 1./math.sqrt(self.weight:size(1))
    end
-   if nn.oldSeed then
-      for i=1,self.templates:size(2) do
-         self.templates:select(2, i):apply(function()
-            return torch.uniform(-stdv, stdv)
-         end)
-      end
+   self.weight:uniform(-stdv, stdv)
+   self.diagCov:fill(1)
+end
+
+function view(res, src, ...)
```

Comment:
it halts for the sequence of instructions so far that i was planning to JIT. Example: if it wanted to inline the function as part of a larger block, it wont be able to.